software/bwameth/align/main.nf

```diff
@@ -0,0 +1,44 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process BWAMETH_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::bwameth=0.2.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bwameth:0.20--py35_0"
+    } else {
+        container "quay.io/biocontainers/bwameth:0.20--py35_0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    path index
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path  "*.version.txt"         , emit: version
+
+    script:
+    def software   = getSoftwareName(task.process)
+    def prefix     = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def read_group = meta.read_group ? "-R ${meta.read_group}" : ""
+    """
+    bwameth.py \\
+        $options.args \\
+        $read_group \\
+        -t $task.cpus \\
+        --reference ${index}/genome.fa \\
```

Comment:
We aren't guaranteed for the index to be named `genome.fa`. [This](https://github.com/nf-core/modules/blob/01c0d335ca57b1b9298a7acd643116917953804a/software/bwa/mem/main.nf#L34) is a more robust solution.

Reply:
Alright, alright :see_no_evil: 
I'm not a big fan of it (looks a bit hacky), so I thought it's enough to ensure the `bwameth_index` module always uses `genome.fa` as filename. It's also the case for iGenomes.

But I don't have strong opinions on it, so here we go :)